conans/client/cache/cache.py

```diff
@@ -155,6 +157,22 @@ def config(self):
             self._config = ConanClientConfigParser(self.conan_conf_path)
         return self._config
 
+    @property
+    def new_config_path(self):
+        return os.path.join(self.cache_folder, "conan_conf.txt")
```

Comment:
I'd prefer this to be decoupled from the cache folder.
otherwise, we introduce a vicious circle here: config has a path to the cache folder, but the config path is relative to the cache folder. it's super confusing.

Reply:
This is not new, the current config (and absolutely everything from the cache) is already loaded this way, because assuming it is in the cache folder, you need the cache folder to locate it. It cannot be decoupled, but maybe I don't understand the suggestion.

Reply:
I'd say it's not always a valid argument "it was always like that" when designing a brand new thing, especially if approach is already known to be problematic.
first, it's logically inconsistent, as the configuration file doesn't belong to cache idiomatically (we are not caching it, we caching binaries and sources).
IMO it belongs to the config directory (which might be the same or might not be the same as cache).

Reply:
It is not an argument of "it was always like that". It is maybe a slightly incorrect naming. Lets rename the ``cache_folder`` to ``home_folder`` which is what it actually is. Conan codebase is referring to the Conan cache and the Conan home folder equivalently, because 97% of the home folder is devoted to the package cache. Basically, the Conan home folder contains the settings.yml, the remotes.json, the conan.db, the profiles plus the recipes and packages cache.

If we agree with that, I am very fine with renaming the Conan codebase from "cache" => "home" when it applies. But the logic implemented in this ``new_config_path()`` is correct and in the right place.